.github/actions/BuildJlm/action.yml

```diff
@@ -78,6 +70,14 @@ runs:
       run: ./configure.sh ${{steps.collect.outputs.jlm_configure_arguments}}
       shell: bash
 
+    - name: "Install HLS dialect dependencies"
+      if: ${{inputs.enable-hls == 'true'}}
+      uses: ./.github/actions/BuildCirct
+
+    - name: "Install MLIR dialect dependencies"
+      if: ${{inputs.enable-mlir == 'true'}}
+      uses: ./.github/actions/BuildMlirDialect
+
```

Comment:
All these can be reverted once you just have the shell.config as not auto-generated file.